### PR TITLE
refactor(): remove deprecated controllers

### PR DESCRIPTION
**Pull request by liamdebeasi**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

v4 had a number of controllers that will be removed in v5. This PR removes them.

## What is the new behavior?

- Removed deprecated controllers from v4

## Does this introduce a breaking change?

- [x] Yes
- [ ] No

### CODE REVIEW

File: core/src/components.d.ts [hunk 1]
```diff
@@ -9,10 +9,8 @@
 import { HTMLStencilElement, JSXBase } from '@stencil/core/internal';
 import {
   ActionSheetButton,
-  ActionSheetOptions,
   AlertButton,
   AlertInput,
-  AlertOptions,
```
Comment (brandyscarney) [inline core/src/components.d.ts]: Don't we need these exposed?

File: core/src/components/action-sheet/test/basic/index.html [hunk 2]
```diff
@@ -57,11 +61,14 @@
   </style>
   <script>
     window.addEventListener('ionActionSheetDidDismiss', function (e) { console.log('didDismiss', e) })
-
+    async function openActionSheet(opts) {
+      const actionSheet = await window.actionSheetController.create(opts);
```
Comment (brandyscarney) [inline core/src/components/action-sheet/test/basic/index.html]:
```suggestion
      const actionSheet = await actionSheetController.create(opts);
```

File: core/src/components/action-sheet/test/basic/index.html [hunk 3]
```diff
@@ -114,7 +119,7 @@
               buttons: [{
                 text: 'Okay',
                 handler: async () => {
-                  await actionSheetElement.dismiss();
+                  await window.actionSheetController.dismiss();
```
Comment (brandyscarney) [inline core/src/components/action-sheet/test/basic/index.html]:
```suggestion
                  await actionSheetController.dismiss();
```

You shouldn't need window on any of these

File: core/src/components.d.ts (hunk 1 above)
Comment (liamdebeasi) [reply core/src/components.d.ts]: Hmm those were automatically removed. Let me dig into why.

File: core/src/components/loading/test/standalone/index.html [hunk 4]
```diff
@@ -10,13 +10,16 @@
   <script src="../../../../../scripts/testing/scripts.js"></script>
   <script nomodule src="../../../../../dist/ionic/ionic.js"></script>
   <script type="module" src="../../../../../dist/ionic/ionic.esm.js"></script></head>
-
+  <script type="module">
+    import { loadingController } from '../../../../dist/ionic/index.esm.js';
+    loadingController = loadingController;
```
Comment (brandyscarney) [inline core/src/components/loading/test/standalone/index.html]:
```suggestion
    window.loadingController = loadingController;
```

**Pull request merged**